lib/phobos/executor.rb

```diff
@@ -1,7 +1,17 @@
 module Phobos
   class Executor
     include Phobos::Instrumentation
-    LISTENER_OPTS = %i(handler group_id topic min_bytes max_wait_time force_encoding start_from_beginning max_bytes_per_partition).freeze
+    LISTENER_OPTS = %i(
+      handler
+      group_id
+      topic
+      min_bytes
+      max_wait_time
+      force_encoding
+      start_from_beginning
+      max_bytes_per_partition
+      backoff
```

Comment:
Down at https://github.com/klarna/phobos/pull/35/files#diff-fd46456a68de2cf7020fc43770bec636R62 in `run_listener(listener)` is where we actually use the exponential backoff we create. And right now, we still use the phobos config base defaults. I think we need to use the config for that listener (if it has custom back off configured).

maybe something like
```ruby
backoff = listener.create_exponential_backoff
```

Otherwise it will sleep for the wrong duration than the user expected.

❤️  if you manage to a test that covers this in https://github.com/klarna/phobos/blob/master/spec/lib/phobos/executor_spec.rb ?